yakut/param/formatter.py

```diff
@@ -94,12 +175,65 @@ def _unittest_formatter() -> None:
         }
     }
     assert (
-        _FORMATTERS["YAML"]()(obj)
-        == """---
+            _FORMATTERS["YAML"]()(obj)
+            == """---
 2345:
   abc:
     def: [123, 456]
   ghi: 789
 """
     )
     assert _FORMATTERS["JSON"]()(obj) == '{"2345":{"abc":{"def":[123,456]},"ghi":789}}'
+    assert _FORMATTERS["TSV"]()(obj) == "123\t456\t789"
+    tsvh_formatter = _FORMATTERS["TSVH"]()
+    # first time should include a header
+    assert tsvh_formatter(obj) == "2345.abc.def.[0]\t2345.abc.def.[1]\t2345.ghi\n123\t456\t789"
+    # subsequent calls shouldn't include a header
+    assert tsvh_formatter(obj) == "123\t456\t789"
+    from decimal import Decimal
+    from math import nan
+
+    obj = {
+        142: {
+            "_metadata_": {
+                "timestamp": {"system": Decimal("1640610921.414715"), "monotonic": Decimal("4522.612870")},
+                "priority": "nominal",
+                "transfer_id": 17,
+                "source_node_id": 21,
+            },
+            "timestamp": {"microsecond": 66711825},
+            "value": {
+                "kinematics": {
+                    "angular_position": {"radian": nan},
+                    "angular_velocity": {"radian_per_second": 375860.15625},
+                    "angular_acceleration": {"radian_per_second_per_second": 0.0},
+                },
+                "torque": {"newton_meter": nan},
+            },
+        }
+    }
+    assert _FORMATTERS["TSV"]()(obj) == "1640610921.414715	4522.612870	nominal	17	21	66711825	nan	375860.15625	0.0	nan"
```

Comment:
The same should be tested against the other two formatters.

Reply:
yes

Reply:
done